src/compiler/moduleNameResolver.ts

```diff
@@ -1536,11 +1564,116 @@ namespace ts {
         return withPackageId(packageInfo, loadNodeModuleFromDirectoryWorker(extensions, candidate, onlyRecordFailures, state, packageJsonContent, versionPaths));
     }
 
+    /* @internal */
+    export function getEntrypointsFromPackageJsonInfo(
+        packageJsonInfo: PackageJsonInfo,
+        options: CompilerOptions,
+        host: ModuleResolutionHost,
+        cache: ModuleResolutionCache | undefined,
+        resolveJs?: boolean,
+    ): string[] | false {

```

Comment:
It’s not so much for the benefit of the caller as for the benefit of this function knowing whether it has computed the result for this `packageJsonInfo`. If `packageJsonInfo.resolvedEntrypoints` is `undefined`, there is no cached value. It stores `false` if it calculates no entrypoints so that it doesn’t recompute it again in the future.